hearthstone/simulator/core/player.py

```diff
@@ -570,3 +570,10 @@ def play_spell(self, index: 'SpellIndex', board_index: Optional['BoardIndex'] =
         spell = self.pop_spell(index)
         self.coins -= spell.cost
         spell.on_play(BuyPhaseContext(self, self.tavern.randomizer), board_index, store_index)
+
+    def swap_hero(self, old_hero: 'Hero', new_hero: 'Hero'):
+        self.hero = new_hero
+        new_hero.secrets.extend(old_hero.secrets)
```

Comment:
It might make more sense to have secrets be a property of the player instead of the hero.  Then you don't need to do this swap.